packages/starksheet-webapp/src/provider/EVMProvider.ts

```diff
@@ -110,6 +119,14 @@ export class EVMProvider implements ChainProvider {
     return this.provider.getTransactionReceipt(hash);
   }
 
+  /**
+   * @inheritDoc
+   */
+  getWorksheetContractByAddress(address: string): WorksheetContract {
+    const abi = chainAbi[this.config.chainType].spreadsheet;
```

Comment:
now that I am reading this, I realize this could be further genericized by making the `ChainProvider` an abstract class rather than an interface, because this code is a duplicate with the StarknetProvider implementation. However we've only used interface implementation so far (and not class inheritance), so maybe overkill to introduce a new concept just for this function (and there is already a lot of refactoring in this PR).

If function duplication increases in the provider this could be a good idea to do yet another refacto tho